core/tag.c

```diff
@@ -148,8 +151,14 @@ instant_t _lf_time(_lf_time_type type) {
 
 ////////////////  Functions declared in tag.h
 
-tag_t lf_tag() {
```

Comment:
It seems more natural to me to leave this function as it was. With your change, there are lots of places where it ends up being called with `NULL` as an argument, which is kind of ugly. Why not reference `self->current_tag` directly with a local time is needed? It's not less readable than `lf_tag(self)`.

Reply:
I see that there is a dependency on whether there currently is an executing reaction. If you want to include that logic as a convenience, then maybe it's better to create a separate function called `lf_local_tag` that takes a reference to the self struct?

Reply:
> I see that there is a dependency on whether there currently is an executing reaction. If you want to include that logic as a convenience, then maybe it's better to create a separate function called `lf_local_tag` that takes a reference to the self struct?

I don't think there is any semantic meaning to a global tag, so I don't think this is a good idea. I think that really there is *only* a local tag. It just so happens that we have had in the past a phenomenon that the local tag is advanced together for all reactors on the same platform.  This is not the case for federated, nor will it be the case for enclaves or LET.

My suggestion is to leave this as is, and document that users should *never* pass NULL to these functions, and that in the future, this will result in an error. I think all the places where NULL is passed now are internal, and to implement enclaves, these will have to replaced with non-NULL calls.